core-processor/src/ext.rs

```diff
@@ -1016,18 +1205,21 @@ impl Externalities for Ext {
     }
 
     fn lock_payload(&mut self, at: u32, len: u32) -> Result<PayloadSliceLock, Self::FallibleError> {
-        let end = at
-            .checked_add(len)
-            .ok_or(FallibleExecutionError::TooBigReadLen)?;
-        self.charge_gas_if_enough(
-            self.context
-                .costs
-                .syscalls
-                .gr_read_per_byte
-                .cost_for(len.into()),
-        )?;
-        PayloadSliceLock::try_new((at, end), &mut self.context.message_context)
-            .ok_or_else(|| FallibleExecutionError::ReadWrongRange.into())
+        self.with_changes(|mutator| {
```

Comment:
it seems this approach with `with_changes` won't work for `gr_read` syscall. It'd discuss that in pm